packages/react-relay/modern/ReactRelayRefetchContainer.js

```diff
@@ -225,6 +234,10 @@ function createContainerWithFragments<TBase: ReactClass<*>>(
         }
         // TODO t15106389: add helper utility for fetching more data
         this._pendingRefetch = null;
+        this._relayContext = {
+          environment: this.context.relay.environment,
+          variables: fragmentVariables
```

Comment:
A fragment container and pagination container only has visibility of ['RootArgument' ](https://github.com/facebook/relay/blob/master/packages/relay-runtime/store/RelayConcreteVariables.js#L44);  only refetch container has 'LocalArgument' and it should keep it to itself just in case it would clash with root ones